### PR TITLE
games-mud/tkmoo: EAPI8, ebuild improvements

### PR DESCRIPTION
**Pull request by mm1ke**
Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Michael Mair-Keimberger <mmk@levelnine.at>

@ionenwks

### CODE REVIEW

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE=""
 
 DEPEND="
-	>=dev-lang/tcl-8.3.3:0=
-	>=dev-lang/tk-8.3.3:0="
-RDEPEND=${DEPEND}
+	dev-lang/tcl:0=
+	dev-lang/tk:0="
```
Comment (ionenwks) [inline games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:20]: The 0 slot isn't needed anymore either.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild [hunk 2]
```diff
@@ -1,44 +1,45 @@
...
+	"${FILESDIR}/${PV}-keys-workaround.patch"
 )
 
 src_compile() {
 	emake \
 		WISH="$(type -P wish)" \
-		TKMOO_LIB_DIR="/usr/$(get_libdir)/${MY_PN}" \
-		TKMOO_BIN_DIR=/usr/bin
+		TKMOO_LIB_DIR="${EPREFIX}/usr/$(get_libdir)/${MY_PN}" \
+		TKMOO_BIN_DIR="${EPREFIX}/usr/bin"
```
Comment (ionenwks) [inline games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:34]: I personally came to prefer replacing those backslash `\` mess by a variable even if it's just for 2-3 lines. i.e.
```
local emakeargs=(
	VAR=val
)
emake "${emakeargs[@]}"
```
`\` is also more error prone, if you forget a backlash it doesn't necessarily fail and bash just sets the value in the environment.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
 	emake \
-		TKMOO_LIB_DIR="${D}/usr/$(get_libdir)/${MY_PN}" \
-		TKMOO_BIN_DIR="${D}/usr/bin" \
+		TKMOO_LIB_DIR="${ED}/usr/$(get_libdir)/${MY_PN}" \
+		TKMOO_BIN_DIR="${ED}/usr/bin" \
 		install
+
 	dodoc README dot.tkmoolightrc bugsmail.txt
 	dosym tkMOO-lite /usr/bin/tkmoo
 	make_desktop_entry tkmoo "tkMOO"
```
Comment (ionenwks) [inline games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:45]: Does this install an icon? (didn't check)

**Review by ionenwks — COMMENTED** [review]
Just a quick look (didn't mean to send previous as a single comment)

I was actually going to do this one tomorrow (next in my list), but hey, now it's one less to do.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild (hunk 3 above)
Comment (mm1ke) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:48]: Unfortunately it has no icon. I've checked the homepage for something but didn't find anything there either.

Comment (ionenwks) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:48]: Just use `applications-games` then, better than using an icon that doesn't exist or making something up from nothing.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild (hunk 1 above)
Comment (ionenwks) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild]: On a side-note, I meant to say only the 0 while keeping :=, however upon closer look this doesn't link with the libraries so the := really isn't needed and this is fine :)

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild [hunk 4]
```diff
@@ -1,44 +1,48 @@
...
-inherit eutils
+EAPI=8
 
-MY_PN=${PN/moo/MOO-light}
-MY_P=${P/moo/MOO-light}
+inherit desktop
+
+MY_PN="${PN/moo/MOO-light}"
+MY_P="${P/moo/MOO-light}"
 DESCRIPTION="MOO Client written in Tcl/Tk"
```
Comment (ionenwks) [inline games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:10]: Tiny nitpick but I prefer a blank line after variables before description.

Also, I find the substitution kind of silly, and done twice for nothing, e.g. /could/
```
MY_PN="tkMOO-light"
MY_P="${MY_PN}-${PV}"
```

However see my next comment, I think `MY_P="tkMOO-light-${PV}"` is all we need here.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild [hunk 5]
```diff
@@ -1,44 +1,48 @@
...
 )
 
 src_compile() {
-	emake \
-		WISH="$(type -P wish)" \
-		TKMOO_LIB_DIR="/usr/$(get_libdir)/${MY_PN}" \
-		TKMOO_BIN_DIR=/usr/bin
+	local myemakeargs=(
+		WISH="$(type -P wish)"
+		TKMOO_LIB_DIR="${EPREFIX}/usr/$(get_libdir)/${MY_PN}"
```
Comment (ionenwks) [inline games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild:33]: Maybe it'd make more sense to use `${EPREFIX}/usr/share/${PN}`, this is just tcl files. Same for src_install.

No real need to use MY_PN here either, should use the package's gentoo name for its directories if possible.

Comment (mm1ke) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild]: Nice suggestion, I've changed the ebuild and got rid of `MY_PN` completely. :)

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild (hunk 1 above)
Comment (mm1ke) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild]: Oh i see, good to know.

File: games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild (hunk 5 above)
Comment (ionenwks) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild]: Still keeping `$(get_libdir)`? I didn't see much point in having tcl script files in lib64, not byte-compiled or anything like that either, it's just arch-independent plain text.

Last thing worth a mention, rest looks good.

Comment (mm1ke) [reply games-mud/tkmoo/tkmoo-0.3.32-r1.ebuild]: Completely oversaw this :D 
Fixed now.

**Review by ionenwks — APPROVED** [review]
Thanks, merging as-is.

**Pull request closed without merging**